### DISCUSSION TITLE
Miscellaneous dev questions

### DISCUSSION THREAD

**Discussion by Henddher** [evennia/evennia#2890] [Evennia Engine Development]
Hi all,

Couple of misc questions:

- I just realized that Coveralls is not running on the branch `develop`. If this is on purpose?
- What's the situation with code formatting? (`isort`, `black`, etc.)? Are we supposed to run it manually?
- What's the situation with `poetry`?

**Comment by Griatch** [comment]
- `coverage` (which is the local thing you run) should be running on `develop`, what makes you say it does not? There is however a strange situation with `coveralls` (the online location to which you can push your `coverage` results), where the develop branch is listed as having much lower percentage coverage (25%)  than what you get when viewing it locally (66%). Not sure why that is, and have had no time to investigate it - it suddenly changed a few months back. 
- I personally run `isort` and `black` (with line-width of 100) automatically for my own code, but I also run it on the code as a whole semi-regularly. I don't require contributors to set these things up, so as to lower the technical threshold of contributing. 
- There is no situation with `poetry`. We use `pip` for now, which is as close to a standard as we have.

**Reply by Henddher** [reply]
Thanks @Griatch!

Are there handy instructions for running `coverage`?

I looked into how to run `coverage` on `django` but it quickly became a rabbit hole (https://django-testing-docs.readthedocs.io/en/latest/coverage.html#configure-django-nose)

**Reply by selberhad** [reply]
You can check out the Github Action that runs it on the repo [here](https://github.com/evennia/evennia/blob/develop/.github/workflows/github_action_test_suite.yml#L125-L138).

Might make sense to do in a ubuntu Docker container so that it's running in the same environment as on Github.

**Reply by Henddher** [reply]
Thank you @selberhad

```shell
 coverage run --source=$PATH_TO_REPO/evennia \
    --omit=*/migrations/*,*/urls.py,*/test*.py,*.sh,*.txt,*.md,*.pyc,*.service 
    $PATH_TO_EVENNIA_VENV/bin/evennia test
    --settings=settings --keepdb
    evennia
coverage report
```
(lifted and adapted for future ref)
